### PR TITLE
OBJ-50 adding validation to objecting entity details screen

### PR DESCRIPTION
**Pull request by SVarallo**
Jira ticket at https://companieshouse.atlassian.net/browse/OBJ-50

### CODE REVIEW

File: src/controllers/objecting.entity.name.controller.ts [hunk 1]
```diff
@@ -1,27 +1,66 @@
...
+import { check, ValidationError, validationResult } from "express-validator";
+import { ErrorMessages } from "../model/error.messages";
+import { createGovUkErrorData, GovUkErrorData } from "../model/govuk.error.data";
 import { OBJECTIONS_COMPANY_NUMBER } from "../model/page.urls";
 import { addObjectionCreateToObjectionSession } from "../services/objection.session.service";
 import { ObjectionCreate } from "../modules/sdk/objections";
 import { Session } from "ch-node-session-handler";
+import { Templates } from "../model/template.paths";
+
+const FULL_NAME_FIELD: string = "fullName";
```
Comment (markpit) [inline src/controllers/objecting.entity.name.controller.ts:11]: when declaring string literal constants you can leave off the : string 
ie. x = "some string", you don't need : string 
x: string = getMessage() you would need it.

Comment (markpit) [reply src/controllers/objecting.entity.name.controller.ts:11]: it's ok as it is, just a best practice thing

File: src/controllers/objecting.entity.name.controller.ts [hunk 2]
```diff
@@ -1,27 +1,66 @@
...
  * @param req
  * @param res
  * @param next
  */
-export const post = (req: Request, res: Response, next: NextFunction) => {
-  const shareIdentityString: string = req.body.shareIdentity;
-  const fullNameValue: string = req.body.fullName;
-  const shareIdentityValue: boolean = shareIdentityString === "yes";
-  const objectionCreate: ObjectionCreate = { shareIdentity: shareIdentityValue, fullName: fullNameValue };
+export const processForm = [...validators, async (req: Request, res: Response, next: NextFunction): Promise<void> => {
```
Comment (markpit) [inline src/controllers/objecting.entity.name.controller.ts:25]: function needs to use an await if you declare it as async, therefore you can change function signature to 
export const processForm = [...validators, (req: Request, res: Response, next: NextFunction) => {

to remove the async and Promise<void> as we are not using await

File: src/routes/routes.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ const renderTemplate = (template: string) => (req: Request, res: Response) => {
 router.get("/", renderTemplate(Templates.INDEX));
 
 router.get(pageURLs.OBJECTING_ENTITY_NAME, renderTemplate(Templates.OBJECTING_ENTITY_NAME));
-router.post(pageURLs.OBJECTING_ENTITY_NAME, objectingEntityNameRoute.post);
+router.post(pageURLs.OBJECTING_ENTITY_NAME, objectingEntityNameRoute.processForm);
```
Comment (markpit) [inline src/routes/routes.ts:31]: could we keep it as post for consistency

**Review by markpit — APPROVED** [review]

**Review by ICoxCH — APPROVED** [review]

**Pull request merged**